// *****************************************************************************
//
// This file is part of a MASA library or program.
// Refer to the included end-user license agreement for restrictions.
//
// Copyright (c) 2010 MASA Group
//
// *****************************************************************************

package MsgsClientToAar;

message MsgPlotRequest {
    message TimeRange {
        required uint32 first_tick = 1;
        required uint32 duration   = 2;
    }
    required uint32    identifier = 1;
    required string    request    = 2;
    optional TimeRange time_range = 3; //!< The request time range
}

message MsgClientToAar {
    message Content {
        optional MsgPlotRequest plot_request = 1;
    }
    optional int32   context = 1;
    required Content message = 2;
}
